mm2src/coins/eth.rs

```diff
@@ -681,6 +891,8 @@ pub struct EthCoinImpl {
     pub(crate) platform_fee_estimator_state: Arc<FeeEstimatorState>,
     /// Config provided gas limits for swap and send transactions
     pub(crate) gas_limit: EthGasLimit,
+    /// Config provided gas limits v2 for swap v2 transactions
+    pub(crate) gas_limit_v2: EthGasLimitV2,
```

Comment:
I think these two fields should be refactored into an enum instead..or tell me why they shouldn't be please

Reply:
We shouldnt use enum for gas limits, as we need to keep both Gas limit versions (for legacy Etomic Swap Contract methods and for Etomic Swap Contracts V2) at the same time.

When we start to support TPU V2 in production, we will have users who didnt do update and still use legacy swaps, in this case other trading side have to do fallback to legacy swap version. 
Another case is that right now not all coins support TPU V2, so trader should use legacy protocol, if in the swap there is a coin which doesnt support TPU V2.

I have [draft pr](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261) for these cases when we need to do legacy swap fallback. It will be r2r after the current pr merge to not to cause conflicts (right now branch requests merge to `eth-maker-tpu-v2`)

Reply:
thank you very much for the detailed explanation.